Develop v2 1

amazzini opened the pull request.

Commit:
PhotonicMedia model enhancements

See "2020-05-04/08 TAPI Virtual Meeting Agenda and Notes", Wed 6/P3

Other minor diagram adjustments
(TapiCommon.uml “visibility” on LayerProtocolQualifier, no actual
changes)

Commit:
The saved Photonic model

Commit:
Removed LocalClass inheritance from ResilienceRoute class

Typos in Resilience Route Priority comment
Notification: Added “DEVICE” to ObjectType
(the EquipmentObjectType enum in eqp model cannot extend ObjectType enum
defined in Notification)
Other minor adjustments checking "model validation"

karthik-sethuraman reviewed (changes requested):
Need to include YANG, TREE and OAS files

Commit:
Merge remote-tracking branch 'TAPI213/develop_v2_1' into develop_v2_1

Commit:
PhotonicMedia: see "2020-06-09 TAPI Meeting notes"

Notification: added DEVICE to ObjectType
VirtualNtw: "schedule" type changed to "tapi-common:time-range"
Connectivity: minor corrections in comments
Adding YANG and TREE modules

Commit:
OAS modules (experimental)

arthurMll commented on YANG/tapi-photonic-media@2020-04-23.yang:
```diff
@@ -714,6 +748,49 @@ module tapi-photonic-media {
         }
         description "none";
     }
+    grouping media-channel-aconnectivity-service-end-point-spec {
```
Typo in the name: media-channel-**_a_**connectivitiy-service-end-point-spec

arthurMll commented on YANG/tapi-photonic-media@2020-04-23.yang:
```diff
@@ -292,6 +304,20 @@ module tapi-photonic-media {
+        description "none";
+    }
+    augment "/tapi-common:context/tapi-connectivity:connectivity-context/tapi-connectivity:connectivity-service/tapi-connectivity:end-point" {
+        container media-channel-aconnectivity-service-end-point-spec {
```
Typo in the name: media-channel-**a**connectivitiy-service-end-point-spec

arthurMll commented:
I have reviewed the .yang .tree files and it seems they do not match. I think the tree files are generated wrong or based in an intermediate version of the YANG files.

Commit:
Adjustments agreed at TAPI call 2020-06-16:

1) supportable-central-frequency-spectrum-band instead of
supportable-lower-central-frequency, supportable-upper-central-frequency

1.1) added keys lower/upper frequency to CentralFrequencyBand type

2) removed redundant association between
MediaChannelConnectivityServiceEndPointSpec and PowerManagementConfigPac

3) renamed (through Papyrus) the YANG/OAS modules which have been
modified with respect to RC1 ones (connectivity, notification, photonic
media, streaming, virtual ntw)

amazzini commented:
Also renamed
- OtsiAConnectivityServiceEndPointSpec as OtsiaConnectivityServiceEndPointSpec
- MediaChannelAConnectivityServiceEndPointSpec as McaConnectivityServiceEndPointSpec

nigel-r-davis commented:
I have just reviewed the photonic media model and yang and have found another name issue following the OtsiAConnect... pattern. There is an OtsiGServerAdaptationPac. This comes out as otsi-gserver.... It ought to be otsig-server..., hence the uml should be OtsigServerAdaptationPac. There are also two forms, Otsia... and OtsiAssembly... in the uml. The OtsiAssemblyConnectionEndPointSpec... ought to be OtsiaConnectionEndPointSpec. 

Unfortunately we also have an longhand media-channel-assembly-spec - this may be arguable because we are using longhand media-channel everywhere else not mc, so the distinction between OTSI and MC is set in general. 

I suggest we stay with media-channel-assembly, but ought to fix the Otsig issue (as it is exactly the same pattern as the Otsia issue we have already fixed. We could consider the OtsiAssembly issue from a consistency perspective, but I am ok to leave it as is.

nigel-r-davis commented:
Have just realized that the two items I commented on are both deprecated.

I have no issue with the PhotonicMedia model as it is.

nigel-r-davis reviewed (approved):
To the depth I have managed to review, I am OK to proceed.

arthurMll reviewed (approved):
Everything seems correct. Approved from my side.
